src/mkdocs_gallery/plugin.py

```diff
@@ -348,11 +348,13 @@ def on_serve(self, server, config, builder):
         def wrap_callback(original_callback):
             def _callback(event):
                 for g in excluded_dirs:
-                    # TODO maybe use fnmatch rather ?
-                    if event.src_path.startswith(g):
+                    try:
+                        Path(event.src_path).relative_to(g)
```

Comment:
The `if` statement seems to have disappeared ? 

Could you please validate that this works on the mkdocs-gallery site ?

- run mkdocs serve
- modify a file in the source dir: it should update the site
- modify a file in the doc/generated/gallery files: it should not trigger a site recreate, since the path is in `excluded_dirs`

Reply:
> The `if` statement seems to have disappeared ?

Yes indeed: in fact the implementation mimics the `is_relative_to` [implementation in pathlib](https://github.com/python/cpython/blob/4cc4fe27890ef73d91c49b1bc0686a50ec704ade/Lib/pathlib.py#L945), which appeared in 3.9

Reply:
> Could you please validate that this works on the mkdocs-gallery site ?

I did, and it works.

Do you know how to implement non-regression tests for this particular use case? This seems tricky...

Reply:
Yes, adding a `is_relative_to(p1, p2)` util function in `utils.py` will be my preferred option.

For now testing this is super-tricky, so manual test on PR is sufficient. Thanks !

Reply:
> Yes, adding a `is_relative_to(p1, p2)` util function in `utils.py` will be my preferred option.

Done. With unit tests